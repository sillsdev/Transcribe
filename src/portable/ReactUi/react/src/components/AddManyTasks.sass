@import "../_variables.sass"

.AddManyTasks
    width: 100%
    height: 100%
    position: absolute
    top: 0
    left: 0
    background-color:transparent
    display: flex
    flex-direction: row
    justify-content: flex-end
    &.rtl
        direction: rtl
        .panel
            padding-right: 0
            &>*
                padding-left: 0
            &>.titleRow
                padding-right: 0px
    .panel
        width: 510px
        height: 100%
        background-color: $base02
        display: flex
        flex-direction: column
        flex: 0 0 auto
        overflow-y: visible
        &>.titleRow
            padding-left: 0
            padding-bottom: 32px
        padding-left: 0
    .titleRow
        display: flex
        flex-direction: row
        padding-top: 32px
        z-index: 2
        .BackLink
            width: 64px
            text-align: center
    .NextAction
        display: flex
        font-family: $sans
        margin: auto
        cursor: pointer
        width: 100px
        padding: 0px 5px
        align-self: center

    .helpText
        color: $base1
        font-family: $sans
        font-size: 16px
        line-height: 28px
        letter-spacing: 0.44px
        padding: 24px 48px
        text-align: center

    .linkText
        color: $blueLink
        &:hover
            cursor: pointer

    .LinkAction
        width: 220px
        margin: 0px

    .browseDiv
        width: 446px
        height: 124px
        display: flex
        border: 1px dotted white
        justify-content: center
        vertical-align: middle
        margin-left: 32px
        margin-right: 32px
        align-items: center
