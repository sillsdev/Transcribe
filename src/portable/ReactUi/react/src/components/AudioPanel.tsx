import * as React from 'react';
import * as actions from '../actions/audioActions';
import { log } from '../actions/logAction';
import * as actions1 from '../actions/taskActions';
import { ITranscriberStrings } from '../model/localize';
import './AudioPanel.sass';
import ControlPane from './ControlPane';
import EditorPane from './EditorPane';
import ProgressPane from './ProgressPane';

interface IProps extends IStateProps, IDispatchProps {
};

class AudioPanel extends React.Component<IProps, object> {
    public render() {
        log("AudioPanel")
        return (<div className="AudioPanel">
                    <div className="ProgressRow">
                        <ProgressPane {...this.props} />
                    </div>
                    <div className="EditorRow">
                        <EditorPane {...this.props} />
                    </div>
                    <div className="ControlRow">
                        <ControlPane {...this.props} />
                    </div>
                </div>
        )
    }
};

interface IStateProps {
    assignedReview: ITask[];
    assignedTranscribe: ITask[];
    direction: string;
    playing: boolean;
    playSpeedRate: number;
    selectedTask: string;
    selectedUser: string;
    jump: number;
    strings: ITranscriberStrings;
    users: IUser[];
};

interface IDispatchProps {
    playStatus: typeof actions.playStatus,
    playSpeedRateChange: typeof actions.playSpeedRateChange;
    jumpChange: typeof actions.jumpChange;
    reportPosition: typeof actions.reportPosition;
    saveTotalSeconds: typeof actions.saveTotalSeconds;
    setSubmitted: typeof actions.setSubmitted;
    showHelp: typeof actions1.showHelp,
};

export default AudioPanel;
