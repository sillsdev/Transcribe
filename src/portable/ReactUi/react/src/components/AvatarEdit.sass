@import "../_variables.sass"

.AvatarEdit
	width: 100%
	height: 100%
	position: absolute
	top: 0
	left: 0
	background-color: transparent
	display: -ms-flexbox
	display: flex
	-ms-flex-direction: row
	flex-direction: row
	-ms-flex-pack: end
	justify-content: flex-end
	&.rtl
        direction: rtl
        .panel
            padding-right: 0
            &>*
                padding-left: 0
                padding-right: 64px
            &>.titleRow
                padding-right: 0px
	.panel
        width: 352px
        height: 100%
        background-color: $base02
        display: flex
        flex-direction: column
        flex: 0 0 auto
        overflow-y: auto
        &>*
            padding-left: 64px
            flex: 0 0 auto
        &>.titleRow
            padding-left: 0
            padding-bottom: 32px
        padding-left: 0
	.titleRow
        display: flex
        flex-direction: row
        padding-top: 32px
        z-index: 2
        .BackLink
            width: 64px
            text-align: center
        .title
            align-self: flex-start
            flex: 1 0 auto
        .copyAction.hide
            display: none
        .copyAction
            display: flex
            flex-direction: column
            align-self: center
            padding: 0 64px
	.data
		z-index: 2
	.data>*>*
		width: 448px
	.dataColumn
		max-width: 392px
	.data>*>.SelectField .dataColumn
		max-width: 448px
	.options
		display: flex
		flex-direction: column
		flex: 1 0 auto
		padding: 70px
		.uploadContent
			position: relative
			overflow: hidden
			display: inline-block
		.btn
			border: 2px solid gray
			color: #002B36
			background-color: #F5CC4C
			padding: 8px 20px
			border-radius: 4px
			font-family: "Noto Sans", "Roboto", sans-serif
			font-size: 14px
			line-height: 16px
			letter-spacing: 0.75px
		.uploadContent input[type=file]
			font-size: 100px
			position: absolute
			left: 0
			top: 0
			opacity: 0
		.zoomContent
			padding-top: 20px
	.action
        background-color: $base02
        display: flex
        flex: 0 0 auto
        z-index: 1
        justify-content: space-between
        flex-direction: row
        border-top: 1px solid $base01
	.IconButtonField
		margin: 0
	#discard label
		width: 144px
input
	color: $base1
	font-family: $sans
	font-style: normal
	font-size: 12px
	line-height: 16px
	letter-spacing: 15%

