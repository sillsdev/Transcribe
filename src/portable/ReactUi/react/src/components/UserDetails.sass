@import "../_variables.sass"

.UserDetails.Modal
    background-color: $base03
    -webkit-filter: blur(5px)
    -moz-filter: blur(5px)
    -o-filter: blur(5px)
    -ms-filter: blur(5px)
    filter: blur(5px)

.UserDetails
    width: 100%
    height: 100%
    position: absolute
    top: 0
    left: 0
    background-color: transparent
    display: flex
    flex-direction: row
    justify-content: flex-end
    &.rtl
        direction: rtl
        .panel
            padding-right: 0
            &>*
                padding-left: 0
                padding-right: 64px
            &>.titleRow
                padding-right: 0px
    .panel
        width: 576px
        height: 100%
        background-color: $base02
        display: flex
        flex-direction: column
        flex: 0 0 auto
        overflow-y: auto
        &>*
            padding-left: 64px
            flex: 0 0 auto
        &>.titleRow
            padding-left: 0
            padding-bottom: 32px
        padding-left: 0
    .closeRow
        display: flex
        flex-direction: row
        align-self: flex-end
        padding: 20px 20px 0 0
        outline: none
        display: none
    .titleRow
        display: flex
        flex-direction: row
        padding-top: 32px
        z-index: 2
        align-items: center
        .BackLink
            width: 64px
            text-align: center
        .copyAction.hide
            display: none
        .copyAction
            display: flex
            flex-direction: column
            align-self: center
            padding: 0 64px
        .titleRow
        display: flex
        flex-direction: row
        padding-top: 32px
        z-index: 2
        .BackLink
            width: 64px
            text-align: center
        .title
            flex: 1 0 auto
        .copyAction.hide
            display: none
        .copyAction
            display: flex
            flex-direction: column
            align-self: center
            padding: 0 64px
    .data
        z-index: 2
    .data>*>*
        width: 448px
        .dataColumn
            max-width: 392px
    .data>*>.SelectField .dataColumn
            max-width: 448px

    .action
        background-color: $base02
        display: flex
        flex: 0 0 auto
        z-index: 1
        justify-content: space-between
        flex-direction: row
        border-top: 1px solid $base01
        & label
            color: $base1
    .IconButtonField
        margin: 0
    #discard label
        width: 144px
    #deleteUserHide
        display: none
    .scrollWindow
        background-color: $base03
        flex: 1 0 auto
        overflow: auto
    .preview
        background-color: $base03
        display: flex
        flex: 1 0 auto
        z-index: 0
        flex-direction: column
        box-shadow: 0px 2px 2px rgba(0,0,0,0.25)
        .LabelCaptionUx
            padding: 24px 0 8px 16px
            color: $yellowLight
    .hide
        visibility: hidden
    .details
        display: flex
        flex-direction: row
        padding: 20px

    .AvatarRow.hide
        display: none
    .AvatarRow
        display: flex
        flex-direction: row
        padding: 24px 0
        .AvatarDetails
            display: flex
            flex-direction: column
            padding: 0 24px
    .AvatarCaption
        display: flex
        flex-direction: column
        color: $base2
        font-family: Roboto
        line-height: 32px
        font-size: 18px
        letter-spacing: 0.444444px
        color: #EEE8D5

    .LabelUx
        padding: 0px

    TextInputFieldUx .TextInputFieldUx
        padding-bottom: 40px

    .none
        display: none
