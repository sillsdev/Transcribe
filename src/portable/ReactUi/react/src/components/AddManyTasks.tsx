import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from '../actions/taskActions';
import { IUserSettingsStrings } from '../model/localize';
import { IProjectSettingsStrings } from '../model/localize';
import { IState } from '../model/state';
import uiDirection from '../selectors/direction';
import userStrings from '../selectors/localize';
import './AddManyTasks.sass';
import BackLink from './controls/BackLink';
import NextAction from './controls/NextAction';

interface IProps extends IStateProps, IDispatchProps {
};

export class AddManyTasks extends React.Component<IProps, object> {

    public AddManyTasks = () => {
        const { addManyTasks, selectedUser, selectedProject } = this.props;
        addManyTasks(selectedUser, selectedProject);
    }

    public doFetch = () => {
        const {fetchTasksOfProject, selectedProject} = this.props;
        fetchTasksOfProject(selectedProject);
    }

    public render() {
        const { direction, strings, strings2 } = this.props;
        const reactStringReplace = require('react-string-replace');

        const links = [ this.batchUploadSpreadsheetHelp, this.ourNamingConventionHelp ];
        const hotText = [ strings2.batchUpload, strings2.ourNamingConvention ];

        return (
            <div className={"AddManyTasks " + (direction && direction === "rtl" ? "rtl" : "ltr")}>
                <div className="panel">
                    <div className="titleRow">
                        <BackLink action={this.doFetch} target="/ProjectSettings" />
                    </div>
                    <div className="browseDiv">
                        <NextAction text={strings.browse.toUpperCase()} target={this.AddManyTasks} type="primary" />
                    </div>
                    <div className="helpText">
                        <span>
                            {reactStringReplace(strings2.learnToAddMany, /\{(\d+)\}/g, (match: string) => (
                                <a key={"a" + match} className="linkText" onClick={links[match]}>{hotText[match]}</a>
                            ))}
                        </span>
                    </div>
                </div>
            </div>
        );
    }

    private batchUploadSpreadsheetHelp = () => {
        this.props.showHelp("Concepts/Spreadsheet_convention.htm")
    }

    private ourNamingConventionHelp = () => {
        this.props.showHelp("Concepts/Folder_and_Files.htm")
    }

}
interface IStateProps {
    direction: string;
    selectedProject: string;
    selectedUser: string;
    strings: IUserSettingsStrings;
    strings2: IProjectSettingsStrings;
};

const mapStateToProps = (state: IState): IStateProps => ({
    direction: uiDirection(state),
    selectedProject: state.tasks.selectedProject,
    selectedUser: state.users.selectedUser,
    strings: userStrings(state, { layout: "userSettings" }),
    strings2: userStrings(state, {layout: "projectSettings"}),
});
interface IDispatchProps {
    addManyTasks: typeof actions.addManyTasks,
    fetchTasksOfProject: typeof actions.fetchTasksOfProject,
    showHelp: typeof actions.showHelp,
};

const mapDispatchToProps = (dispatch: any): IDispatchProps => ({
    ...bindActionCreators({
        addManyTasks: actions.addManyTasks,
        fetchTasksOfProject: actions.fetchTasksOfProject,
        showHelp: actions.showHelp,
    }, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(AddManyTasks);
