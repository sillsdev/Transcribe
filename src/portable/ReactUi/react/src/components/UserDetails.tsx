import * as React from 'react';
import Avatar from 'react-avatar';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom'
import { bindActionCreators } from 'redux';
import * as actions3 from '../actions/avatarActions';
import { log } from '../actions/logAction';
import * as actions from '../actions/taskActions';
import * as actions2 from '../actions/userActions';
import { IProjectSettingsStrings, IUserSettingsStrings } from '../model/localize';
import { IState } from '../model/state';
import uiDirection from '../selectors/direction';
import userStrings from '../selectors/localize';
import currentProject from '../selectors/project';
import AvatarLink from './controls/AvatarLink';
import BackLink from './controls/BackLink';
import AnchorHelp from './ui-controls/AnchorHelp';
import IconButtonField from './ui-controls/IconButtonField';
import ImageField from './ui-controls/ImageField';
import LabelCaptionUx from './ui-controls/LabelCaptionUx';
import LabelUx from './ui-controls/LabelUx';
import RadioListField from './ui-controls/RadioListField';
import TextField from './ui-controls/TextField';
import './UserDetails.sass';

interface IProps extends IStateProps, IDispatchProps {
    history: {
        location: {
            pathname: string;
        }
    }
};

const initialState = {
    avatarUrl: "",
    discard: false,
    imageFile: "",
    name: "",
    otherProjects: [],
    role: "Transcriber",
    roles: "",
    scrollHeight: 0,
    selectedValue: "",
}

export class UserDetails extends React.Component<IProps, typeof initialState> {
    public state = { ...initialState };
    private original: typeof initialState;
    private userId: string;
    private roleListDef: string[];
    private roleListLoc: string[];
    private imageRef: React.RefObject<ImageField>;
    private scrollRef: React.RefObject<HTMLDivElement>;

    constructor(props: IProps) {
        super(props)
        this.updateUserName = this.updateUserName.bind(this);
        this.updateImageFile = this.updateImageFile.bind(this);
        this.imageRef = React.createRef();
        this.scrollRef = React.createRef();
        this.updateRoles = this.updateRoles.bind(this);
        this.discard = this.discard.bind(this);
        this.deleteUser = this.deleteUser.bind(this);
        const { popupUser, users, strings } = this.props;
        this.roleListDef = ['Admin', 'Reviewer', 'Transcriber'];
        this.roleListLoc = [strings.admin, strings.reviewer, strings.transcriber];
        this.userId = this.props.history.location.pathname.indexOf("NewUser") > 0 ? "" : popupUser;
        
        if (this.userId && this.userId !== "") {
            const user: IUser = users.filter((u: IUser) => u.username.id === this.userId)[0];
            if(user !== undefined)
            {
                this.state.name = user.displayName
                if (user.username.avatarUri !== undefined) {
                    this.props.saveAvatar({data: user.username.avatarUri, uri: user.username.avatarUri});
                    this.state.imageFile = user.username.avatarUri && user.username.avatarUri.replace("/api/images/", "");
                }
                const roleCount = user.role.length
                if (roleCount === 3) {
                    this.state.role = strings.admin;
                }
                else if (roleCount === 2) {
                    this.state.role = strings.reviewer + " + " + strings.transcriber;
                }
                else if (roleCount === 1) {
                    const userRole = this.roleListLoc[this.roleListDef.indexOf(user.role[0].charAt(0).toUpperCase() + user.role[0].slice(1))];
                    if (userRole === strings.transcriber) {
                        this.state.role = strings.transcriber;
                    }
                    else if (userRole === strings.reviewer) {
                        this.state.role = strings.reviewer;
                    }
                    else {
                        this.state.role = strings.transcriber;
                    }
                }
            }
            else {
                this.state = { ...initialState, role: this.roleListLoc[2], name: this.userId }
            }
            
        } else {
            this.props.saveAvatar({data: "/assets/smile.svg", uri: "/assets/smile.svg"})
            this.state = { ...initialState, role: this.roleListLoc[2] }
        }

        this.original = { ...this.state };
    }

    public componentDidMount () {
        const scrollLocation = this.scrollRef && this.scrollRef.current && this.scrollRef.current.offsetTop
            ? this.scrollRef.current.offsetTop
            : 0;
        this.setState({scrollHeight: window.innerHeight - scrollLocation - 52})
    }
    public render() {
        const { avatar, avatarUri, deleted, direction, strings, strings2, project, popupUser, setUserAvatar, users } = this.props;
        const { discard, name } = this.state;
        const save = () => this.save(this);
        log("UserDetails")
        if (deleted || discard) {
            return (<Redirect to="/ProjectSettings" />)
        }
        setUserAvatar();
        const historyPath = this.props.history.location.pathname;
        const newUser = historyPath.indexOf("NewUser") > 0;
        this.userId = historyPath.indexOf("NewUser") > 0 ? "" : popupUser;
        const user: IUser = users.filter((u: IUser) => u.username.id === this.userId)[0];
        const projectAvatar = user && user.project ? (
            <AvatarLink id={project.id}
                name={(project.guid === undefined || project.guid === "" || (project.sync !== undefined && !project.sync)) && project.name? project.name: project.id}
                size="48"
                target="/main"
                avatarShape={false}
                uri={project.uri !== undefined ? project.uri : ""} />) : "";
        const userPos = historyPath.indexOf("User") + 4;
        let settingsStyle = this.props.history.location.pathname.length > userPos? " Modal": ""
        settingsStyle = direction? settingsStyle + " " + direction: settingsStyle;
        return (
            <div className={"UserDetails" + settingsStyle}>
                <div className="panel">
                    <div className="titleRow">
                        <BackLink action={save} target="/ProjectSettings" />
                        <div className="title">
                            <LabelCaptionUx name={strings.userDetails} type="H3" />
                        </div>
                        <div className="anchorHelp">
                                <AnchorHelp id="ProjSettingsHelp" onClick={this.ShowUserDetailsHelp} />
                        </div>
                    </div>
                    <div className="data">
                        <div><TextField id="id1"
                            autofocus={true}
                            caption={strings2.name}
                            inputValue={this.state.name}
                            onChange={this.updateUserName} /></div>
                        <div><ImageField id="id2"
                            caption={strings2.imageFile}
                            inputValue={avatar === avatarUri? this.state.imageFile: avatarUri}
                            fromPath={historyPath} onChange={this.updateImageFile}
                            ref={this.imageRef} /></div>
                    </div>
                    <div className="scrollWindow" ref={this.scrollRef} style={{maxHeight: this.state.scrollHeight}}>
                        <div className={name !== ""? "preview": "none"}>
                            <LabelCaptionUx name={strings.preview} type="small" />
                            <div className={"AvatarRow" + (name !== "" ? "" : " hide")}>
                                <Avatar className="OnHover"
                                    name={name} key={name + "Avatar"}
                                    src={avatar}
                                    size={64}
                                    round={true} />
                                <div className="AvatarDetails">
                                    <div className="AvatarCaption">{name}</div>
                                    <div><LabelUx name={this.state.role} /></div>
                                </div>
                            </div>
                        </div>
                        <div className="details">
                            <div className="results">
                                <div className="resultsRight">
                                    <div className="resultsRightBox">

                                        <div className="rolesBox">
                                            <LabelUx name={strings2.role} />
                                            <RadioListField
                                                options={this.roleListLoc}
                                                selected={this.state.role}
                                                onChange={this.updateRoles} />
                                        </div>
                                        <div className={projectAvatar !== ""? "OtherProjectsBox": "none"}>
                                            <LabelUx name={strings.otherProjects} />
                                            {projectAvatar}
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div className="action">
                        <IconButtonField id="discard"
                            caption={strings.discardChanges}
                            imageUrl="CancelIcon.svg"
                            onClick={this.discard} />
                        <IconButtonField id={"deleteUser" + (newUser ? "Hide" : "")}
                            caption={strings.deleteUser}
                            imageUrl="RejectIcon.svg"
                            onClick={this.deleteUser} />
                    </div>
                </div>
            </div>
        )
    }

    public handleChange(event: any) {
        this.setState({ selectedValue: event.target.value });
    }

    private ShowUserDetailsHelp = () => {
        this.props.showHelp("Edit user details")
    }

    private discard() {
        this.setState({ ...this.state, discard: true })
    }

    private saveValue(updates: string[], tag: string, val: string | null) {
        updates.push(tag + '=' + encodeURIComponent(val != null ? val : ""))
    }

    private updateUserName(newName: string) {
        this.setState({ ...this.state, name: newName })
    }

    private updateImageFile(newImage: string) {
        this.setState({ imageFile: newImage })
        if(newImage.length === 0) {
            this.setState({ avatarUrl: "" })
        }
    }

    private updateRoles(newroles: string) {
        this.setState({ roles: newroles, role: newroles })
    }

    private deleteUser() {
        const { deleteUser, popupUser } = this.props;
        deleteUser(popupUser);
    }

    private save(ctx: UserDetails) {
        const { selectedProject, updateUser } = this.props;

        const updates = Array<string>();
        if (this.state.name !== this.original.name) {
            this.saveValue(updates, "name", this.state.name);
        }

        if (this.state.roles !== this.original.roles) {
            const userRole = this.roleListDef[this.roleListLoc.indexOf(this.state.roles)];
            this.saveValue(updates, "role", userRole);
        }

        if (this.state.avatarUrl !== this.original.avatarUrl || this.original.avatarUrl.length > 0) {
            this.saveValue(updates, "avatarUri", this.state.avatarUrl);
        }

        const img = (this.state.avatarUrl === "" && this.props.avatar && this.props.avatar.indexOf("smile") < 0)?
            this.props.avatar: ""

        if (updates.length > 0 || img !== "") {
            const query = '&' + updates.join('&');
            // tslint:disable-next-line:no-console
            console.log("/api/updateUser?user=" + this.userId, '&project=' + selectedProject + query);
            updateUser(this.userId, selectedProject, query, {img});
        }
    }
}

interface IStateProps {
    avatar: string;
    avatarUri: string;
    deleted: boolean;
    direction: string;
    project: IProject;
    popupUser: string;
    strings: IProjectSettingsStrings;
    strings2: IUserSettingsStrings;
    users: IUser[];
    selectedParatextProject: string;
    selectedProject: string;
};

const mapStateToProps = (state: IState): IStateProps => ({
    avatar: state.avatar.data,
    avatarUri: state.avatar.uri,
    deleted: state.users.deleted,
    direction: uiDirection(state),
    popupUser: state.users.selectedPopupUser,
    project: currentProject(state),
    selectedParatextProject: state.paratextProjects.selectedParatextProject,
    selectedProject: state.tasks.selectedProject,
    strings: userStrings(state, { layout: "projectSettings" }),
    strings2: userStrings(state, { layout: "userSettings" }),
    users: state.users.users,
});

interface IDispatchProps {
    fetchUsers: typeof actions2.fetchUsers;
    selectPopupUser: typeof actions2.selectPopupUser,
    selectTask: typeof actions.selectTask;
    updateUser: typeof actions2.updateUser;
    deleteUser: typeof actions2.deleteUser;
    saveAvatar: typeof actions3.saveAvatar;
    setUserAvatar: typeof actions3.setUserAvatar;
    showHelp: typeof actions.showHelp,
};
const mapDispatchToProps = (dispatch: any): IDispatchProps => ({
    ...bindActionCreators({
        deleteUser: actions2.deleteUser,
        fetchUsers: actions2.fetchUsers,
        saveAvatar: actions3.saveAvatar,
        selectPopupUser: actions2.selectPopupUser,
        selectTask: actions.selectTask,
        setUserAvatar: actions3.setUserAvatar,
        showHelp: actions.showHelp,
        updateUser: actions2.updateUser,
    }, dispatch),
});
export default connect(mapStateToProps,
    mapDispatchToProps
)(UserDetails);