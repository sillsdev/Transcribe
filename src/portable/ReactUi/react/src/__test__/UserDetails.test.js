/* jshint esversion: 6 */
import React from 'react';
import expect from 'expect';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MemoryRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import sinon from 'sinon';
import {UserDetails} from '../components/UserDetails';

configure({ adapter: new Adapter() });

// Snapshot for UserDetails
describe('>>>Control: UserDetails --- Snapshot',()=>{
    const minProps = {
        avatar: "/api/images/smile.svg",
        deleted: false,
        direction: "ltr",
        history: {
            location: {
                pathname: "/ProjectSettings/User"
            }
        },
        popupUser: "Erik",
        project: {
            id: "ztt",
            lang: "fr",
            task: []
        },
        selectedParatextProject: "",
        selectedProject: "ztt",
        setUserAvatar: () => {return},
        strings: {
            "admin": "Admin",
            "deleteUser": "Delete User",
            "discardChanges": "Discard Changes",
            "otherProjects": "Other Projects",
            "preview": "Preview",
            "reviewer": "Reviewer",
            "transcriber": "Transcriber",
            "userDetails": "User Details",
        },
        strings2: {
            "imageFile": "Image File",
            "name": "Name",
            "role": "Role",
        },
        users: [],
    };

    it('+++capturing Snapshot of UserDetails', () => {
        const renderedValue =  renderer.create(<Router><UserDetails {...minProps} snapShotTest={true}/></Router>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    });
});
//*******************************************************************************************************
describe('>>>Control: UserDetails', () => {
    let wrapper;
    const minProps = {
        avatar: "/api/images/smile.svg",
        deleted: false,
        direction: "ltr",
        history: {
            location: {
                pathname: "/ProjectSettings/User"
            }
        },
        popupUser: "Erik",
        project: {
            id: "ztt",
            lang: "fr",
            task: []
        },
        selectedParatextProject: "",
        selectedProject: "ztt",
        setUserAvatar: () => {return},
        strings: {
            "admin": "Admin",
            "deleteUser": "Delete User",
            "discardChanges": "Discard Changes",
            "otherProjects": "Other Projects",
            "preview": "Preview",
            "reviewer": "Reviewer",
            "transcriber": "Transcriber",
            "userDetails": "User Details",
        },
        strings2: {
            "imageFile": "Image File",
            "name": "Name",
            "role": "Role",
        },
        users: [],
    };

    beforeEach(() => {
        wrapper = shallow(<UserDetails {...minProps} />);
    })

    it('+++ renders without exploding', () => {
        expect(wrapper.length).toEqual(1);
    });
});
