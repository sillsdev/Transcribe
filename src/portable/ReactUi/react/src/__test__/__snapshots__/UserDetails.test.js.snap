// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Control: UserDetails --- Snapshot +++capturing Snapshot of UserDetails 1`] = `
<div
  className="UserDetails ltr"
>
  <div
    className="panel"
  >
    <div
      className="titleRow"
    >
      <div
        className="BackLink"
        id="BackLink"
      >
        <a
          href="/ProjectSettings"
          onClick={[Function]}
        >
          ❮
        </a>
      </div>
      <div
        className="title"
      >
        <span
          className="LabelCaptionUx H3 label label-default"
        >
          User Details
        </span>
      </div>
      <div
        className="anchorHelp"
      >
        <div
          className="AnchorHelp"
          id="ProjSettingsHelp"
        >
          <span>
            <a
              className="linkText"
              onClick={[Function]}
            >
              <img
                alt="?"
                src="/assets/QuestionGrey.svg"
              />
            </a>
          </span>
        </div>
      </div>
    </div>
    <div
      className="data"
    >
      <div>
        <div
          className="TextField"
          id="id1"
        >
          <div
            className="bodyRow"
          >
            <div
              className="dataColumn"
            >
              <label
                className="Caption CaptionGreen"
              >
                Name
              </label>
              <input
                autoFocus={true}
                className="DataColor"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Name"
                readOnly={undefined}
                type="text"
                value="Erik"
              />
            </div>
            <div
              className="clearButton"
              onClick={[Function]}
            >
              <img
                alt="X"
                src="/assets/Clear.svg"
              />
            </div>
          </div>
          <div
            className="Message BorderTopGreen"
          >
            <label>
              
            </label>
          </div>
        </div>
      </div>
      <div>
        <div
          className="ImageField"
          id="id2"
        >
          <div
            className="bodyRow"
          >
            <div
              className="dataColumn"
              onClick={[Function]}
            >
              <label
                className="Caption CaptionGreen"
              >
                
              </label>
              <label
                className="fileInput CaptionGrey"
                data-content="Image File"
                htmlFor="Upload"
              />
            </div>
            <div
              className="Hide"
              onClick={[Function]}
            >
              <img
                alt="X"
                src="/assets/Clear.svg"
              />
              <a
                className="pencil"
                href="/ProjectSettings/User/avatar/PopupUser"
                onClick={[Function]}
              >
                ✐
              </a>
            </div>
          </div>
          <div
            className="Message BorderTopGreen"
          >
            <label>
              
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      className="scrollWindow"
      style={
        Object {
          "maxHeight": 716,
        }
      }
    >
      <div
        className="preview"
      >
        <span
          className="LabelCaptionUx small label label-default"
        >
          Preview
        </span>
        <div
          className="AvatarRow"
        >
          <div
            className="OnHover sb-avatar"
            onClick={undefined}
            style={
              Object {
                "borderRadius": "100%",
                "display": "inline-block",
                "fontFamily": "Helvetica, Arial, sans-serif",
                "height": "64px",
                "verticalAlign": "middle",
                "width": "64px",
              }
            }
          >
            <div
              className="OnHover sb-avatar__text"
              style={
                Object {
                  "background": undefined,
                  "borderRadius": "100%",
                  "color": "#FFF",
                  "height": "64px",
                  "lineHeight": "initial",
                  "textAlign": "center",
                  "textTransform": "uppercase",
                  "width": "64px",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "table",
                    "height": "100%",
                    "tableLayout": "fixed",
                    "width": "100%",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "display": "table-cell",
                      "fontSize": "100%",
                      "verticalAlign": "middle",
                      "whiteSpace": "nowrap",
                    }
                  }
                >
                  <span />
                </span>
              </div>
            </div>
          </div>
          <div
            className="AvatarDetails"
          >
            <div
              className="AvatarCaption"
            >
              Erik
            </div>
            <div>
              <span
                className="LabelUx label label-default"
              >
                Transcriber
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="details"
      >
        <div
          className="results"
        >
          <div
            className="resultsRight"
          >
            <div
              className="resultsRightBox"
            >
              <div
                className="rolesBox"
              >
                <span
                  className="LabelUx label label-default"
                >
                  Role
                </span>
                <div
                  className="RadioListField"
                >
                  <div
                    className="bodyRow"
                  >
                    <div
                      className="dataColumn"
                    >
                      <form>
                        <fieldset
                          className="radioFieldSet"
                          id="radioGroup"
                        >
                          <div>
                            <input
                              checked={false}
                              className="DataColor"
                              name="radioGroup"
                              onChange={[Function]}
                              type="radio"
                              value="Admin"
                            />
                            <label
                              className="labelRadioCaption"
                            >
                              Admin
                               
                            </label>
                          </div>
                          <div>
                            <input
                              checked={false}
                              className="DataColor"
                              name="radioGroup"
                              onChange={[Function]}
                              type="radio"
                              value="Reviewer"
                            />
                            <label
                              className="labelRadioCaption"
                            >
                              Reviewer
                               
                            </label>
                          </div>
                          <div>
                            <input
                              checked={true}
                              className="DataColor"
                              name="radioGroup"
                              onChange={[Function]}
                              type="radio"
                              value="Transcriber"
                            />
                            <label
                              className="labelRadioCaption"
                            >
                              Transcriber
                               
                            </label>
                          </div>
                        </fieldset>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="none"
              >
                <span
                  className="LabelUx label label-default"
                >
                  Other Projects
                </span>
                
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="action"
    >
      <div
        className="IconButtonField"
        id="discard"
        onClick={[Function]}
      >
        <img
          alt="D"
          className="CaptionImage"
          src="/assets/CancelIcon.svg"
        />
        <label
          className="Caption"
        >
          DISCARD CHANGES
        </label>
      </div>
      <div
        className="IconButtonField"
        id="deleteUser"
        onClick={[Function]}
      >
        <img
          alt="D"
          className="CaptionImage"
          src="/assets/RejectIcon.svg"
        />
        <label
          className="Caption"
        >
          DELETE USER
        </label>
      </div>
    </div>
  </div>
</div>
`;
